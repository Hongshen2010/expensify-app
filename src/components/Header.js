import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expense records</h1>
        <NavLink exact to="/" activeClassName="is-active">Home </NavLink>
        <NavLink to="/create" activeClassName="is-active">create </NavLink>
        <NavLink to="/help" activeClassName="is-active">help</NavLink>
    </header>
);

export default Header;