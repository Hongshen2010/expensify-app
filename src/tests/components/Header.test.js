import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('Header rendering test', () => {
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toBe('Expensify');
});