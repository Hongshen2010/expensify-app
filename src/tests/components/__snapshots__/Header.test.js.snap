// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering test 1`] = `
<header>
  <h1>
    Expense records
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    create 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    help
  </NavLink>
</header>
`;
